src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java

```diff
@@ -35,115 +34,118 @@
-			final String plackupExe,
-			final String abstractHandlePSGIpath,
-			final String handleManagerPSGIpath,
-			final String handleManagerAllowedUser,
-			final MySQLController mysql,
+			final MongoController mongo,
 			final String shockHost,
 			final AuthToken shockAdminToken,
 			final String perl5lib,
 			final Path rootTempDir,
-			URL authServiceURL)
+			URL authURL, 
+			final String handleAdminRole)
 			throws Exception {
-		
-		authServiceURL = new URL(authServiceURL.toString() + "/Sessions/Login");
-		
-		checkExe(plackupExe, "plackup");
-		checkFile(abstractHandlePSGIpath, "Abstract Handle Service PSGI");
-		checkFile(handleManagerPSGIpath, "Handle Manager PSGI");
+				
 		tempDir = makeTempDirs(rootTempDir, "HandleServiceController-",
 				new LinkedList<String>());
 		
-		setUpHandleServiceMySQLTables(mysql.getClient());
-		
 		handleServicePort = findFreePort();
 		
-		File hsIniFile = createHandleServiceDeployCfg(mysql, shockHost, authServiceURL);
-		
-		/*
-		crusherofheads@icrushdeheads:~$ export PERL5LIB=/kb/deployment/lib
-		crusherofheads@icrushdeheads:~$ export KB_DEPLOYMENT_CONFIG=/kb/deployment/deployment.cfg
-		crusherofheads@icrushdeheads:~$ plackup /kb/deployment/lib/AbstractHandle.psgi
-		2014/07/27 23:26:42 15811 reading config from /kb/deployment/deployment.cfg
-		2014/07/27 23:26:42 15811 using http://localhost:7044 as the default shock server
-		{"attribute_indexes":[""],"contact":"shock-admin@kbase.us","documentation":"http://localhost:7044/wiki/","id":"Shock","resources":["node"],"type":"Shock","url":"http://localhost:7044/","version":"0.8.16"}DBI connect('hsi;host=localhost','hsi',...) failed: Access denied for user 'hsi'@'localhost' (using password: YES) at /kb/deployment/lib/Bio/KBase/AbstractHandle/AbstractHandleImpl.pm line 67
-		Cannot read config file /etc/log/log.conf at /kb/deployment/lib/Bio/KBase/Log.pm line 282.
-		HTTP::Server::PSGI: Accepting connections at http://0:5000/
-		
-		--port for port
-		 */
-		
-		ProcessBuilder handlepb = new ProcessBuilder(plackupExe, "--port",
-				"" + handleServicePort, abstractHandlePSGIpath)
+		File hsIniFile = createHandleServiceDeployCfg(mongo, shockHost, authURL,
+				shockAdminToken, handleAdminRole);
+		String lib_dir = "lib";
+		downloadSourceFiles(tempDir, lib_dir);
+
+		String lib_dir_path = tempDir.resolve(lib_dir).toAbsolutePath().toString();
+		ProcessBuilder handlepb = new ProcessBuilder("uwsgi", "--http", 
+				":" + handleServicePort, "--wsgi-file",
+				"AbstractHandle/AbstractHandleServer.py", "--pythonpath", lib_dir_path)
 				.redirectErrorStream(true)
 				.redirectOutput(tempDir.resolve("handle_service.log").toFile());
 		Map<String, String> env = handlepb.environment();
 		env.put("PERL5LIB", perl5lib);
-		env.put("KB_DEPLOYMENT_CONFIG", hsIniFile.toString());
+		env.put("KB_DEPLOYMENT_CONFIG", hsIniFile.getAbsolutePath().toString());
 		env.put("KB_SERVICE_NAME", HANDLE_SERVICE_NAME);
+		env.put("KB_AUTH_TOKEN", shockAdminToken.getToken());
```

Comment:
Why is this in the environment?

Reply:
I want to pass token in config so that my helper class can have token. Otherwise, I guess I can pass ctx['token'] to each method call. 
https://github.com/kbase/handle_service2/blob/develop/lib/AbstractHandle/AbstractHandleImpl.py#L38

Reply:
I don't understand how this is supposed to work... KB_AUTH_TOKEN will be set by the env when the service starts up, right? So if you use that token in service methods it won't correspond with the user that's actually calling the method. Maybe I don't understand what it's being used for?

Reply:
👍 I got what you meant. removed `KB_AUTH_TOKEN` from environ.